apps/system/js/app_chrome.js

```diff
@@ -326,6 +340,11 @@
     }
   };
 
+  AppChrome.prototype.onPin = function ac_onPin() {
```

Comment:
Nit: calling this onPin might be a bit confusing because the "pinning" happens when you press the pin button. Maybe onPinClick or pinPreview or something?